spec/controllers/api/v1/application_users_controller_spec.rb

```diff
@@ -46,6 +45,34 @@
     end
   end
 
+  describe "find by username returns" do
+    it "a single result when username matches" do
+      api_get :find_by_username, untrusted_application_token, parameters: { username: 'foo_bb' }
+      expect(response.code).to eq('200')
+      expected_response = [{
+        id: bob_brown.application_users.first.id,
+        user: {
+          id: bob_brown.id,
+          username: bob_brown.username,
+          first_name: bob_brown.first_name,
+          last_name: bob_brown.last_name
+        }, unread_updates: 0
+      }].to_json
+      expect(response.body).to eq(expected_response)
+    end
+    it "raises not found when when not found" do
+      expect {
+        api_get :find_by_username, untrusted_application_token, parameters: { username: 'foo' }
+      }.to raise_error(ActiveRecord::RecordNotFound)
+    end
+    it "only finds users belonging to the requesting application" do
+      expect {
+        api_get :find_by_username, trusted_application_token, parameters: { username: 'billy_20' }
```

Comment:
Can this test one application looking for another application's user?  "billy_20" doesn't exist at all.